env/build_test.go

```diff
@@ -18,87 +20,135 @@ func TestBuildEnv(t *testing.T) {
 }
 
 func testBuildEnv(t *testing.T, when spec.G, it spec.S) {
-	when("#NewBuildEnv", func() {
-		it("includes expected vars", func() {
-			benv := env.NewBuildEnv([]string{
-				"CNB_STACK_ID=some-stack-id",
-				"HOSTNAME=some-hostname",
-				"HOME=some-home",
-				"HTTPS_PROXY=some-https-proxy",
-				"https_proxy=some-https-proxy",
-				"HTTP_PROXY=some-http-proxy",
-				"http_proxy=some-http-proxy",
-				"NO_PROXY=some-no-proxy",
-				"no_proxy=some-no-proxy",
-				"NOT_INCLUDED=not-included",
-				"PATH=some-path",
-				"LD_LIBRARY_PATH=some-ld-library-path",
-				"LIBRARY_PATH=some-library-path",
-				"CPATH=some-cpath",
-				"PKG_CONFIG_PATH=some-pkg-config-path",
+	var (
+		mockController *gomock.Controller
+		platform       *testmock.MockPlatform
+	)
+	it.Before(func() {
+		mockController = gomock.NewController(t)
+		platform = testmock.NewMockPlatform(mockController)
+	})
+
+	when("#NewDetectEnv", func() {
+		it("always excludes CNB_ASSETS", func() {
+			benv := env.NewDetectEnv([]string{
+				"CNB_ASSETS=some-assets-path",
 			})
 			out := benv.List()
 			sort.Strings(out)
-			expectedVars := []string{
-				"CNB_STACK_ID=some-stack-id",
-				"CPATH=some-cpath",
-				"HOME=some-home",
-				"HOSTNAME=some-hostname",
-				"HTTPS_PROXY=some-https-proxy",
-				"HTTP_PROXY=some-http-proxy",
-				"LD_LIBRARY_PATH=some-ld-library-path",
-				"LIBRARY_PATH=some-library-path",
-				"NO_PROXY=some-no-proxy",
-				"PATH=some-path",
-				"PKG_CONFIG_PATH=some-pkg-config-path",
-			}
+			var expectedVars []string
```

Comment:
This could be made terser as suggested here: https://github.com/buildpacks/lifecycle/pull/591/files#r635441542

Reply:
Whoops sorry missed that

Reply:
Sorry I should have said "as also suggested" ...I wasn't able to make a new suggestion because some of the lines didn't change...